trackman/public/views.py

```diff
@@ -0,0 +1,382 @@
+                           results=results['results'])
+
+
+@bp.route('/playlists/charts/dj/<int:dj_id>/artists')
+def charts_artists_dj(dj_id):
+    results = DJArtistCharts().get(dj_id)
+
+    if request.wants_json():
+        return jsonify(results)
+
+    return render_template('public/chart_artists_dj.html',
+                           dj=results['dj'],
+                           results=results['results'])
+
+
+@bp.route('/playlists/charts/tracks')
+@bp.route('/playlists/charts/tracks/<string:period>')
+@bp.route('/playlists/charts/tracks/<string:period>/<int:year>')
+@bp.route('/playlists/charts/tracks/<string:period>/<int:year>/<int:month>')
+def charts_tracks(period=None, year=None, month=None):
+    if period == 'dj' and year is not None:
+        return redirect(url_for('.charts_tracks_dj', dj_id=year))
+
+    results = TrackCharts().get(period, year, month)
+
+    if request.wants_json():
+        return jsonify(results)
+
+    return render_template('public/chart_tracks.html',
+                           start=results['start'],
+                           end=results['end'],
+                           results=results['results'])
+
+
+@bp.route('/playlists/charts/dj/<int:dj_id>/tracks')
+def charts_tracks_dj(dj_id):
+    results = DJTrackCharts().get(dj_id)
+
+    if request.wants_json():
+        return jsonify(results)
+
+    return render_template('public/chart_tracks_dj.html',
+                           dj=results['dj'],
+                           results=results['results'])
+
+
+@bp.route('/playlists/charts/dj/spins')
+def charts_dj_spins():
+    results = DJSpinCharts().get()
+
+    if request.wants_json():
+        return jsonify(results)
+
+    return render_template('public/chart_dj_spins.html',
+                           results=results['results'])
+
+
+@bp.route('/playlists/charts/dj/vinyl_spins')
+def charts_dj_vinyl_spins():
+    results = DJVinylSpinCharts().get()
+
+    if request.wants_json():
+        return jsonify(results)
+
+    return render_template('public/chart_dj_vinyl_spins.html',
+                           results=results['results'])
+# }}}
+
+
+@bp.route('/playlists/set/<int:set_id>')
+def playlist(set_id):
+    results = Playlist().get(set_id)
+
+    if request.wants_json():
+        results.update({
+            'archives': [a[0] for a in results['archives']],
+            'tracks': [
+                tracklog_full_serialize(t) for t in results['tracks']],
+        })
+        return jsonify(results)
+
+    return render_template('public/playlist.html',
+                           archives=results['archives'],
+                           djset=results,
+                           tracklogs=results['tracks'])
+
+
+@bp.route('/playlists/track/<int:track_id>')
+def playlists_track(track_id):
+    results = PlaylistTrack().get(track_id)
+
+    if request.wants_json():
+        results['added'] = str(results['added'])
+        results['plays'] = [tracklog_serialize(tl) for tl in results['plays']]
```

Comment:
rewrite to use tracklog_full_serialize

Reply:
I'll just remove this instead; we don't need to support this.